app/javascript/chat/util.js

```diff
@@ -1,35 +1,40 @@
 import 'intersection-observer';
 import { sendKeys } from './actions';
 
+export function getCsrfToken() {
+  const element = document.querySelector(`meta[name='csrf-token']`);
+
+  return element !== null ? element.content : undefined;
+}
+
+const getWaitOnUserDataHandler = ({ resolve, reject, waitTime = 20 }) => {
+  let totalTimeWaiting = 0;
+
+  return function waitingOnUserData() {
+    if (totalTimeWaiting === 3000) {
+      reject(new Error("Couldn't find user data on page."));
+      return;
+    }
+
+    const csrfToken = getCsrfToken(document);
+    const { user } = document.body.dataset;
+
+    if (user && csrfToken !== undefined) {
+      const currentUser = JSON.parse(user);
+
+      resolve({ currentUser, csrfToken });
+      return;
+    }
+
+    totalTimeWaiting += waitTime;
+    setTimeout(waitingOnUserData, waitTime);
```

Comment:
The previous implementation was using `setInterval` which could have bogged things down as setInterval just fires after n ms. regardless of whether or not what's in it is completed. With `setTimeout` things can run within it and once they're complete it can be called again. Habing said that, this still doesn't seem like the best approach, but the current implementation is looking for DOM attributes being updated.

[MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) is an option as you can configure it to only listen for specific attributes' changes, but I think there is probably a better approach than setting these values on DOM attributes. However, since this PR was initally just to rename and move some files, it seems out of scope of this PR.